hasher-matcher-actioner/hmalib/common/tests/test_timebucket.py

```diff
@@ -152,3 +156,39 @@ def test_destroy_empty_buffer(self):
                 [],
                 "Destroy method should not have executed as the buffer is empty",
             )
+
+    @freeze_time("2012-08-13 14:04:00")
+    def test_squash_content(self):
+        with tempfile.TemporaryDirectory() as td:
+            with freeze_time(datetime.datetime.now()) as frozen_datetime:
+
+                VALUE_1 = 5
+                VALUE_2 = 10
+                VALUE_3 = 3
+                for i in range(VALUE_1):
+                    for i in range(VALUE_2):
+                        sample = TimeBucketizer(
+                            datetime.timedelta(minutes=1), td, "hasher", str(i)
+                        )
+                        for i in range(VALUE_3):
+                            content = "".join(
+                                random.choice(string.ascii_lowercase) for _ in range(10)
+                            )
+                            sample.add_record(HashRecord(content, str(i)))
+                        sample.force_flush()
+                    frozen_datetime.tick(delta=datetime.timedelta(minutes=1))
+
+                TimeBucketizer.squash_content(
+                    datetime.datetime.now(), "hasher", td, datetime.timedelta(minutes=1)
+                )
+
+                records = TimeBucketizer.get_records(
+                    datetime.datetime.now() - datetime.timedelta(minutes=7),
+                    datetime.datetime.now(),
+                    "hasher",
+                    td,
+                    datetime.timedelta(minutes=1),
+                    HashRecord,
+                )
+
+                self.assertEqual(len(records), VALUE_1 * VALUE_2 * VALUE_3)
```

Comment:
We had discussed offline that you can test for equality of the records and not just their counts.

Try `self.assertEqual(records, expected_records)` where expected_records is a list you append to along `sample.add_record()`.

To allow for the lists to be out of order, you can use the [`TestCase.assertCountEqual`](https://docs.python.org/3.2/library/unittest.html#unittest.TestCase.assertCountEqual) function.